### PR TITLE
deprecate tidb config enable-replica-selector-v2

### PR DESCRIPTION
**Pull request by crazycs520**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

deprecated tidb config `enable-replica-selector-v2`

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- Other reference link(s): 
  - https://github.com/tikv/client-go/pull/1265
  - https://github.com/pingcap/tidb/pull/53698

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-configuration-file.md [hunk 1]
```diff
@@ -751,11 +751,11 @@ opentracing.reporter 相关的设置。
 + 默认值：60
 + 单位：秒
 
-### `enable-replica-selector-v2` <span class="version-mark">从 v8.0.0 版本开始引入</span>
+### `enable-replica-selector-v2`（已废弃）
```
Comment (joey-yez) [inline tidb-configuration-file.md:754]: 不需要修改 title

Comment (crazycs520) [reply tidb-configuration-file.md]: done.

File: tidb-configuration-file.md [hunk 2]
```diff
@@ -755,7 +755,7 @@ opentracing.reporter 相关的设置。
 
 > **警告：**
 >
-> 该配置项可能会在未来版本中废弃，**不要修改该配置**。
+> 自 v8.2.0 起，该配置项已被废弃。
```
Comment (lilin90) [inline tidb-configuration-file.md:758]:
```suggestion
> 从 v8.2.0 开始，该配置项被废弃。TiKV 发送 RPC 请求时，默认使用新版本的 Region 副本选择器。
```

**Review by lilin90 — APPROVED** [review]
/lgtm

**Review by joey-yez — APPROVED** [review]

File: tidb-configuration-file.md [hunk 3]
```diff
@@ -755,7 +755,7 @@ opentracing.reporter 相关的设置。
 
 > **警告：**
 >
-> 该配置项可能会在未来版本中废弃，**不要修改该配置**。
+> 从 v8.2.0 开始，该配置项被废弃。TiKV 发送 RPC 请求时，默认使用新版本的 Region 副本选择器。
```
Comment (crazycs520) [inline tidb-configuration-file.md:758]:
```suggestion
> 从 v8.2.0 开始，该配置项被废弃。发送 RPC 请求时，默认使用新版本的 Region 副本选择器。
```

File: tidb-configuration-file.md [hunk 4]
```diff
@@ -755,7 +755,7 @@ opentracing.reporter 相关的设置。
 
 > **警告：**
 >
-> 该配置项可能会在未来版本中废弃，**不要修改该配置**。
+> 从 v8.2.0 开始，该配置项被废弃。发送 RPC 请求时，默认使用新版本的 Region 副本选择器。
```
Comment (crazycs520) [inline tidb-configuration-file.md:758]:
```suggestion
> 从 v8.2.0 开始，该配置项被废弃。给 TiKV 发送 RPC 请求时，默认使用新版本的 Region 副本选择器。
```

**Review by hfxsd — APPROVED** [review]

**Review by lilin90 — COMMENTED** [review]
/approve

**Pull request merged**